src/main/java/graphql/schema/idl/TypeDefinitionRegistry.java

```diff
@@ -453,13 +454,13 @@ public boolean isObjectType(Type type) {
     }
 
     /**
-     * Returns the list of object types that implement the given interface type
+     * Returns the list of object and interface types that implement the given interface type
      *
      * @param targetInterface the target to search for
      * @return the list of object types that implement the given interface type
      */
-    public List<ObjectTypeDefinition> getImplementationsOf(InterfaceTypeDefinition targetInterface) {
-        List<ObjectTypeDefinition> objectTypeDefinitions = getTypes(ObjectTypeDefinition.class);
+    public List<ImplementingTypeDefinition> getImplementationsOf(InterfaceTypeDefinition targetInterface) {
```

Comment:
this is a breaking change. Should we consider adding another one and keeping this one for use object implementations?

Reply:
Yep, just noticed that this is public API. I'll create a new method.